### DISCUSSION TITLE
Upgrading TypeScript to 5.2.2 causes next build to fail with DeprecationError

### DISCUSSION THREAD

**Discussion by JamesSingleton** [vercel/next.js#56195] [Help]
### Summary

In some of my projects when upgrading TypeScript (all other deps like `eslint-config-next`, and `eslint` are upgraded to the latest at the time of writing this).

```
-    "typescript": "^5.1.6"
+    "typescript": "^5.2.2"
```

I get the DeprecationError for `originalKeywordKind`.
<img width="806" alt="Screenshot 2023-09-28 at 8 53 01 PM" src="https://github.com/vercel/next.js/assets/21000200/61fcb812-5486-45d8-968f-622fb7484e18">

If I revert back to 5.1.6 it builds just fine.

**Comment by icyJoseph** [comment]
What next.js version do you have? Did you try canary?

**Reply by JamesSingleton** [reply]
@icyJoseph this is on "next": "^13.5.3", installing `"next": "^13.5.4-canary.10",` and upgrading TypeScript results in the same

**Comment by mrkosima** [comment]
**TL;DR:**

Just run:
```
npm update @typescript-eslint/parser
```

---
I encountered the same issue, and it turns out the problem lies with the `eslint-config-next@14.0.0` package, which specifies the following dependency in its `package.json`:

```json
"@typescript-eslint/parser": "^5.4.2 || ^6.0.0"
```

In my `package-lock.json`, I found that `@typescript-eslint/parser` was locked to version **5.54.1**, which can lead to a 'DeprecationError: 'originalKeywordKind'` error.

To resolve this issue, you can update the `@typescript-eslint/parser` package to version **6.x.x** by running the following command:

```
npm update @typescript-eslint/parser
```

Make sure to verify that your `package-lock.json` now contains `@typescript-eslint/parser` at version 6.x.x.

**Reply by JamesSingleton** [reply]
So I ended getting this error again on another package I was upgrading. However, I have `@typescript-eslint/parser@6.14.0` installed and `eslint-config-next@14.0.4` installed. What is _odd_ is that it seems to be `"next/core-web-vitals"` that causes it for me. If I just use `"next"` in the `.eslintrc.json` my linting passes.

**Reply by mrkosima** [reply]
To double check the installed versions of a dependency, you can use `npm ls` within the project, e.g.

```sh
npm ls @typescript-eslint/parser
```

It could be that some packages use different major versions of the dependency, which can cause such issues.
